deepCR/training.py

```diff
@@ -0,0 +1,217 @@
+""" module for training new deepCR-mask models
+"""
+import torch
+from tqdm import tqdm as tqdm
+from deepCR.unet import WrappedModel, UNet2Sigmoid
+import matplotlib.pyplot as plt
+import torch.nn as nn
+import torch.optim as optim
+from torch.autograd import Variable
+from torch.utils.data import DataLoader
+import numpy as np
+import datetime
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from deepCR.util import maskMetric
+from deepCR.dataset import dataset
+
+__all__ = 'train'
+
+class train():
+
+    def __init__(self, image, mask, ignore=None, sky=None, name='model', hidden=32, gpu=False, epoch=50, batch_size=16, lr=0.005, aug_sky=[0, 0], save_after=0, plot_every=10, directory='./'):
+        """ This is the class for training deepCR-mask.
+        :param image: np.ndarray (N*W*W) training data: image array with CR.
```

Comment:
you might consider making this either an (N*W*W)  numpy array OR a pytorch Dataset (with a load from disk capability) -- a preloaded numpy array will limit the user to their available RAM.